src/profile-logic/profile-data.js

```diff
@@ -150,6 +150,167 @@ export function getSampleCallNodes(
   });
 }
 
+// This function assumes that the path isn't empty
+export function getFuncIndex(path: CallNodePath): IndexIntoFuncTable {
+  return path[path.length - 1];
+}
+
+// Returns the JS information for a specific stack
+export function getJsInformationForStack(
+  stackIndex: IndexIntoStackTable,
+  { stackTable, frameTable, stringTable }: Thread
+): string {
+  const frameIndex = stackTable.frame[stackIndex];
+  const jsEngineInformationStrIndex = frameTable.implementation[frameIndex];
+  const jsInformation =
+    jsEngineInformationStrIndex === null
+      ? 'interpreter'
+      : stringTable.getString(jsEngineInformationStrIndex);
+
+  return jsInformation;
+}
+
+type TimingsForSpecificPath = {|
+  selfTime: Milliseconds, // time spent in this path excluding children
+  totalTime: Milliseconds, // time spent in this path including children
+  // information about the JavaScript engine used for this path
+  jsEngineInformations: { [key: string]: Milliseconds } | null,
```

Comment:
I use `implementationBreakdown` because I'm adding `native`

Reply:
Actually I used `breakdownByImplementation`, in an object, like this:
```
{
      forPath: {
        selfTime: { value: 1, breakdownByImplementation: { native: 1 } },
        totalTime: { value: 1, breakdownByImplementation: null },
      },
      forFunc: {
        selfTime: { value: 1, breakdownByImplementation: null },
        totalTime: { value: 2, breakdownByImplementation: null },
      },
      rootTime: 4,
    });
```

I did it this way so that we can easily add the timings for `totalTime` too, as well as easily add other breakdowns (especialy `breakdownByCategory`)